src/algebra/ordered_group.lean

```diff
@@ -93,482 +101,683 @@ instance ordered_comm_group.has_exists_mul_of_le (α : Type u)
+lemma lt_inv_iff_mul_lt_one' : a < b⁻¹ ↔ b * a < 1 :=
+(mul_lt_mul_iff_left' b).symm.trans $ by rw mul_inv_self
+
+end has_mul_lt_mul_left
+
+end left
+
+section right
+
+section covariant_le_right
+variables [covariant_class α α (function.swap (*)) (≤)] {a b c : α}
+
+@[simp, to_additive]
+lemma mul_le_mul_iff_right' (c : α) : a * c ≤ b * c ↔ a ≤ b :=
+⟨λ h, by simpa using mul_le_mul_right' h c⁻¹, λ h, mul_le_mul_right' h _⟩
 
 @[to_additive]
-lemma inv_inv_of_one_lt (h : 1 < a) : a⁻¹ < 1 :=
-have a⁻¹ < 1⁻¹, from inv_lt_inv' h,
-by rwa one_inv at this
+lemma mul_inv_le_iff'' : a * c⁻¹ ≤ b ↔ a ≤ b * c :=
+by rwa [← mul_le_mul_iff_right' c, inv_mul_cancel_right]
 
-@[to_additive neg_of_neg_pos]
-lemma inv_of_one_lt_inv (h : 1 < a⁻¹) : a < 1 :=
-have 1⁻¹ < a⁻¹, by rwa one_inv,
-lt_of_inv_lt_inv this
+@[to_additive neg_nonpos_iff]
+lemma inv_le_one_iff' : a⁻¹ ≤ 1 ↔ 1 ≤ a :=
+by rw [← mul_le_mul_iff_right' a, mul_left_inv, one_mul]
 
-@[to_additive neg_pos_of_neg]
-lemma one_lt_inv_of_inv (h : a < 1) : 1 < a⁻¹ :=
-have 1⁻¹ < a⁻¹, from inv_lt_inv' h,
-by rwa one_inv at this
+@[to_additive right.nonpos_of_neg_nonneg]
+lemma right.le_one_of_one_le_inv : 1 ≤ a⁻¹ ↔ a ≤ 1 :=
+by rw [← mul_le_mul_iff_right' a, mul_left_inv, one_mul]
 
 @[to_additive]
-lemma le_inv_of_le_inv (h : a ≤ b⁻¹) : b ≤ a⁻¹ :=
-begin
-  have h := inv_le_inv' h,
-  rwa inv_inv at h
-end
+lemma le_inv_iff_mul_le_one : a ≤ b⁻¹ ↔ a * b ≤ 1 :=
+by rw [← mul_le_mul_iff_right' b, mul_left_inv]
+@[simp, to_additive]
+lemma div_le_div_iff_right (c : α) : a / c ≤ b / c ↔ a ≤ b :=
+by { rw [div_eq_mul_inv, div_eq_mul_inv], exact mul_le_mul_iff_right' c⁻¹ }
+
+@[simp, to_additive sub_nonneg]
+lemma one_le_div'' : 1 ≤ a / b ↔ b ≤ a :=
+by rw [← mul_le_mul_iff_right' b, one_mul, div_eq_mul_inv, inv_mul_cancel_right]
+
+alias sub_nonneg ↔ le_of_sub_nonneg sub_nonneg_of_le
+
+@[simp, to_additive sub_nonpos]
+lemma div_le_one'' : a / b ≤ 1 ↔ a ≤ b :=
+by rw [← mul_le_mul_iff_right' b, one_mul, div_eq_mul_inv, inv_mul_cancel_right]
+
+alias sub_nonpos ↔ le_of_sub_nonpos sub_nonpos_of_le
 
 @[to_additive]
-lemma inv_le_of_inv_le (h : a⁻¹ ≤ b) : b⁻¹ ≤ a :=
-begin
-  have h := inv_le_inv' h,
-  rwa inv_inv at h
-end
+lemma le_div_iff_mul_le_right : b ≤ c / a ↔ b * a ≤ c :=
+by rw [← mul_le_mul_iff_right' a, div_eq_mul_inv, inv_mul_cancel_right]
 
 @[to_additive]
-lemma lt_inv_of_lt_inv (h : a < b⁻¹) : b < a⁻¹ :=
-begin
-  have h := inv_lt_inv' h,
-  rwa inv_inv at h
-end
+lemma le_div_iff_mul_le : a ≤ c / b ↔ a * b ≤ c :=
+by rw [← mul_le_mul_iff_right' b, div_eq_mul_inv, inv_mul_cancel_right]
+
+alias le_sub_iff_add_le ↔ add_le_of_le_sub_right le_sub_right_of_add_le
 
 @[to_additive]
-lemma inv_lt_of_inv_lt (h : a⁻¹ < b) : b⁻¹ < a :=
-begin
-  have h := inv_lt_inv' h,
-  rwa inv_inv at h
-end
+lemma div_le_iff_le_mul_left : a / b ≤ c ↔ a ≤ c * b :=
+by rw [← mul_le_mul_iff_right' b, div_eq_mul_inv, inv_mul_cancel_right]
+
+alias le_sub_iff_add_le_right ↔ add_le_of_le_sub_left le_sub_left_of_add_le
```

Comment:
Did you mean to change the meaning of `add_le_of_le_sub_left` and `le_sub_left_of_add_le` here? The old meanings were
```
alias le_sub_iff_add_le' ↔ add_le_of_le_sub_left le_sub_left_of_add_le
```
IMO swapping around names of lemmas in the same PR as refactoring their typeclasses is a bad idea!

Reply:
I do not understand `alias` very well, so I had tried to leave this untouched... except that I moved the lemma and then thought that I would redefine it with a "math_eq" statement.  l now think that I restored it to its original name and could revert some more files.  diff decreasing!